gammapy/modeling/models/spatial.py

```diff
@@ -1310,7 +1310,10 @@ class PiecewiseNormSpatialModel(SpatialModel):
     tag = ["PiecewiseNormSpatialModel", "piecewise-norm"]
 
     def __init__(self, coords, norms=None, interp="lin", **kwargs):
-        self._coords = coords
+        self._coords = coords.copy()
```

Comment:
What is the purpose of copying the coords?

Reply:
to avoid the modification of the input coords